src/GitVersionCore.Tests/ConfigProviderTests.cs

```diff
@@ -228,18 +234,51 @@ public void VerifyAliases()
         propertiesMissingAlias.ShouldBeEmpty();
     }
 
-    [Test]
-    public void WarnOnExistingGitVersionConfigYamlFile()
+    [TestCase(DefaultRepoPath)]
+    [TestCase(DefaultWorkingPath)]
+    public void WarnOnExistingGitVersionConfigYamlFile(string path)
     {
-        SetupConfigFileContent(string.Empty, "GitVersionConfig.yaml");
+        SetupConfigFileContent(string.Empty, ConfigurationProvider.ObsoleteConfigFileName, path);
 
-        var s = string.Empty;
-        Action<string> action = info => { s = info; };
+        var logOutput = string.Empty;
+        Action<string> action = info => { logOutput = info; };
         Logger.SetLoggers(action, action, action);
 
-        ConfigurationProvider.Provide(repoPath, fileSystem);
+        ConfigurationProvider.Verify(workingPath, repoPath, fileSystem);
 
-        s.Contains("'GitVersionConfig.yaml' is deprecated, use 'GitVersion.yml' instead.").ShouldBe(true);
+        var configFileDeprecatedWarning = string.Format("{0}' is deprecated, use '{1}' instead", ConfigurationProvider.ObsoleteConfigFileName, ConfigurationProvider.DefaultConfigFileName);
+        logOutput.Contains(configFileDeprecatedWarning).ShouldBe(true);
+    }
+
+    [TestCase(DefaultRepoPath)]
+    [TestCase(DefaultWorkingPath)]
+    public void WarnOnAmbigousConfigFilesAtTheSameProjectRootDirectory(string path)
+    {
+        SetupConfigFileContent(string.Empty, ConfigurationProvider.ObsoleteConfigFileName, path);
+        SetupConfigFileContent(string.Empty, ConfigurationProvider.DefaultConfigFileName, path);
+
+        var logOutput = string.Empty;
+        Action<string> action = info => { logOutput = info; };
+        Logger.SetLoggers(action, action, action);
+
+        ConfigurationProvider.Verify(workingPath, repoPath, fileSystem);
+
+        var configFileDeprecatedWarning = string.Format("Ambigous config files at '{0}'", path);
+        logOutput.Contains(configFileDeprecatedWarning).ShouldBe(true);
+    }
+
+    [TestCase(ConfigurationProvider.DefaultConfigFileName, ConfigurationProvider.DefaultConfigFileName)]
+    [TestCase(ConfigurationProvider.DefaultConfigFileName, ConfigurationProvider.ObsoleteConfigFileName)]
+    [TestCase(ConfigurationProvider.ObsoleteConfigFileName, ConfigurationProvider.DefaultConfigFileName)]
+    [TestCase(ConfigurationProvider.ObsoleteConfigFileName, ConfigurationProvider.ObsoleteConfigFileName)]
+    public void ThrowsExceptionOnAmbigousConfigFileLocation(string repoConfigFile, string workingConfigFile)
+    {
+        SetupConfigFileContent(string.Empty, repoConfigFile, repoPath);
+        SetupConfigFileContent(string.Empty, workingConfigFile, workingPath);
+
+        Should.Throw<WarningException>(() => {
```

Comment:
Can you please assert that the returned `WarningException` has an expected message, so we know it was thrown as expected?